src/key.rs

```diff
@@ -996,6 +996,7 @@ impl str::FromStr for Keypair {
     type Err = Error;
 
     #[allow(unused_variables, unreachable_code)] // When built with no default features.
+    #[allow(clippy::diverging_sub_expression)] // Because of panic.
```

Comment:
I disagree, we should just disable `FromStr` impl if neither `global-context` nor `alloc` is available.

Reply:
I can do that, will do it as a separate PR because its unrelated.

Reply:
Can we put it in front of this one?

Reply:
Sure, will do.

Reply:
Yeah, oops, we should've filed an issue for this. The `panic` thing was there to avoid a major version bump at some point in the past but we should've fixed it properly as soon as we were ready to do a major one. (Maybe we need a new label for this? "Must be addressed by next major rev" or something?)

Reply:
I don't see how that was helpful since the code was broken anyway under that configuration, it just blew up during runtime rather than compile time.

Reply:
Only if the user actually called the function (and didn't catch the panic).

If we'd changed things to fail at compile time then previously-compiling code would have stopped compiling.

As for "why change the code at all then" it was to fix #491, in #492.